src/helpers/internal_viewport.ts

```diff
@@ -72,23 +72,29 @@ export class InternalViewport {
       this.sheetId,
       lastRow
     );
+
     const leftColIndex = this.searchHeaderIndex("COL", lastColEnd - this.width, 0);
     const leftColSize = this.getters.getColSize(this.sheetId, leftColIndex);
     const leftRowIndex = this.searchHeaderIndex("ROW", lastRowEnd - this.height, 0);
     const topRowSize = this.getters.getRowSize(this.sheetId, leftRowIndex);
 
-    const width =
-      lastColEnd -
-      this.offsetCorrectionX +
-      (this.canScrollHorizontally
-        ? Math.max(DEFAULT_CELL_WIDTH, Math.min(leftColSize, this.width - lastColSize))
-        : 0);
-    const height =
-      lastRowEnd -
-      this.offsetCorrectionY +
-      (this.canScrollVertically
-        ? Math.max(DEFAULT_CELL_HEIGHT + 5, Math.min(topRowSize, this.height - lastRowSize))
-        : 0);
+    let width = lastColEnd - this.offsetCorrectionX;
+    if (this.canScrollHorizontally) {
+      width += Math.max(
+        DEFAULT_CELL_WIDTH,
+        Math.min(leftColSize, this.width - lastColSize),
+        this.width - width
+      );
+    }
```

Comment:
I'd add a comment to explain what all those computation mean. W/o prior knowledge of the viewport, it's really hard to get what's going on.